src/components/ContentSidebar/SkillsSidebar.js

```diff
@@ -4,22 +4,163 @@
  * @author Box
  */
 
-import React from 'react';
+import * as React from 'react';
 import { FormattedMessage } from 'react-intl';
+import noop from 'lodash/noop';
+import LoadingIndicator from 'box-react-ui/lib/components/loading-indicator/LoadingIndicator';
 import messages from '../messages';
 import SidebarContent from './SidebarContent';
+import APIContext from '../APIContext';
 import SidebarSkills from './Skills/SidebarSkills';
+import API from '../../api';
 
-type Props = {
+type PropsWithoutContext = {
     file: BoxItem,
-    getPreviewer: Function,
-    onSkillChange: Function
+    getPreviewer: Function
 };
 
-const SkillsSidebar = ({ file, getPreviewer, onSkillChange }: Props) => (
-    <SidebarContent title={<FormattedMessage {...messages.sidebarSkillsTitle} />}>
-        <SidebarSkills file={file} getPreviewer={getPreviewer} onSkillChange={onSkillChange} />
-    </SidebarContent>
-);
+type Props = {
+    api: API
+} & PropsWithoutContext;
+
+type State = {
+    cards?: Array<SkillCard>,
+    errors: NumberBooleanMap
+};
+
+class SkillsSidebar extends React.PureComponent<Props, State> {
+    state: State = {
+        errors: {}
+    };
+
+    componentDidMount() {
+        const { api, file }: Props = this.props;
+        api.getMetadataAPI(false).getSkills(
```

Comment:
Not sure if I am missing something, but not seeing ```getSkills``` (or other metadata api's) in the metadata API. Did you forget to commit it?

Also, I think this should be separated into its own function, and may need to be called again in ```componentWillReceiveProps``` if the file changes

Reply:
I think the diff is hidden since its a lot
https://github.com/box/box-ui-elements/pull/467/files#diff-0c9696951d5577b5baf9b3c5cab4721f